src/ts/deploy.ts

```diff
@@ -0,0 +1,88 @@
+import { utils } from "ethers";
+import { ethers } from "hardhat";
+import { DeployResult, DeploymentsExtension } from "hardhat-deploy/types";
+
+export const contractNames = {
+  authenticator: "GPv2AllowListAuthentication",
+  settlement: "GPv2Settlement",
+};
+
+/**
+ * The salt used when deterministically deploying smart contracts.
+ */
+export const salt = utils.formatBytes32String("dev");
```

Comment:
nit: I think a constant should have SCREAMING_CASE
```suggestion
export const SALT = utils.formatBytes32String("dev");
```

Also, I suspect we can change this salt to something else later. I would suggest:
```suggestion
export const SALT = utils.solidityHash(["string"], ["Gnosis Protocol v2"]);
```

Reply:
Case is fixed.
The hash must not be unique nor have any particular property, so using `0x00...00` would be perfectly fine.
I went for "dev" to make sure that we can migrate during testing without fearing that we might be deploying over the address of the final version of the contract but using it for testing in that network (with owner changes, for example). It's not really so important, since this issue can be easily solved by slightly changing the deployed contract (e.g., the name of a function), but it also doesn't cost much to add this extra safeguard.